FaunaDB.Client/Types/RefParser.cs

```diff
@@ -0,0 +1,76 @@
+﻿namespace FaunaDB.Types
+{
+    public static class RefParser
+    {
+        public static RefV Parse(Value value)
+        {
+            return value.GetOption(Field.At("@ref")).Match(
```

Comment:
I think if you know the api version ahead of time when creating the client you can break this down into two parsers: one for version 2.0, and one for version 2.1.

The parsing then becomes more assertive than attempted.